tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -665,7 +665,9 @@ static void process_wait() {
     DEBUG_STATUS('P', 'W', 'D');
 
     if (clear_count) {
-        *sem_addr = 0;
+        uint32_t neg_sem_val = -(*sem_addr);
+        noc_semaphore_inc(get_noc_addr_helper(my_noc_xy, get_semaphore(upstream_sync_sem)), neg_sem_val);
```

Comment:
This looks like not the correct change, and it is responsible for hangs.  I might have not understood it when I added it, and put the wrong semaphore/address in this call.   I think I should just pass `DISPATCH_MESSAGE_ADDR` as the address instead of `get_semaphore(upstream_sync_sem)` ? Trying that.

Reply:
That looks right to me